H-954: Migrate OSS edition of hCore (#42)
hashintel/labs: apps/sim-core/README.md

It uses a legacy version of hEngine which is no longer maintained, separate from the primary [HASH Engine] (**hEngine**) found in this repository.

## Project Status

> `sim-core` is currently undergoing a migration process away from our internal infrastructure and toward this open-source, self-hosted version.  While we're making it available at this time so that our users can preserve and run their preesting simulations, additional migraiton work is ongoing.  Those upcoming tasks include:

Reviewer: ```suggestion
**hCore** is currently in the process of transitioning from being closed-source and hosted on our internal infrastructure towards being a free, open-source IDE available to self-host. While we're making it available at this time so that users can continue to work with and run existing simulations, additional migration work is ongoing, and we'll be changing the way simulations are created in the future. Upcoming tasks in 'phase one' of the migration include:
```

- [ ] Re-enabling 'new simulation' creation flow
- [ ] Allow for "Project Export" functionality in the development environment
- [ ] Direct GitHub integration for simulation management and storage
- [X] Removal of legacy UI elements which no longer function

> While we work toward completing this migration project, please be mindful of the software's current limitations.
> ## Limitations

Reviewer: ```suggestion
While we work toward completing phase one, please be mindful of the software's current limitations.

Phase two of our simulation migration involves letting users create, work with and run [HASH Core] and [HASH Engine] simulations in the [HASH] application directly.

## Limitations
```

> The sim-core engine is strongly limited to being a run-only environment. Current recommended use is as follows:

Reviewer: ```suggestion
In its present form, the version of hCore published here is for the most part limited to providing a run-only environment for simulations. Current recommended use is as follows:
```

> 1) Run sim-core on localhost and view in your browser

Reviewer: ```suggestion
1) Run hCore (this `apps/sim-core` project) on `localhost` and view it in your browser
```

> 2) To open a simulation, use the 'import' functionality and target a .zip file containing a previously exported simluation.

Reviewer: ```suggestion
2) To open a simulation, use the 'import' functionality and target a `.zip` file containing a previously exported simulation. _This can be downloaded from a project's hIndex listing page._
```

3) You can now run and edit this simulation, however file storage is simply maintained within your browser (using `localstorage`), and changes you make will only be preserved within this web browser.
4) You can use the 'recent projects' menu to switch between other projects that you have imported.
5) To experiment with an example project, import an example project .zip file from the `example_projects` folder.

Please exercise caution if authoring work inside the self-hosted environment because any simulations you author are **not being presreved** outside of the browser environment.  These limitations will lift as the project status goals above are accomplished.

## Installation

> Before running this software, your environment will need to have installed modern versions of:

Reviewer: ```suggestion
Before running this software, your environment will need to have installed modern versions of:
```

[Node](https://nodejs.org/en/), [Rust](https://www.rust-lang.org/learn/get-started), and [Yarn](https://yarnpkg.com/lang/en/).

With these in place, you must use yarn to install wasm-pack:
```sh
yarn global add wasm-pack
```

To verify your installation, from the `sim-core` directory run
```sh
node -v
yarn -v
rustup default
```
If these commands output version numbers, you're all set.
For the first build, simply run:
```sh
yarn
```

### Supported Environments

> The above requirements are all cross-platform, and as such sim-core can build and run in modern Windows, OSX, and Ubuntu, as well as within common VM's and containers.

Reviewer: ```suggestion
The required dependencies above are available (and consistent) across platforms. hCore can be built and run in modern Windows, macOS, and Ubuntu Linux environments, as well as within common VMs and containers.
```

> ## Running `sim-core`

Reviewer: ```suggestion

## Running `sim-core`
```

> To run the sim-core IDE, run:

Reviewer: ```suggestion
To run hCore, run:
```

```sh
yarn serve:core
```

> This will compile the application and host it for you at a default location of [localhost:8080](http://localhost:8080)

Reviewer: ```suggestion
This will compile the application and host it for you at a default location of [`localhost:8080`](http://localhost:8080).
```

### Development and Troubleshooting

> See the README in [`packages/core`](https://github.com/hashintel/labs/tree/main/apps/sim-core/packages/core) for more details

Reviewer: ```suggestion
See the README in [`packages/core`](https://github.com/hashintel/labs/tree/main/apps/sim-core/packages/core) for more details.
```

### Repository Structure

> Several different packages in this repository are orchistrated as yarn workspaces.  The key packages are:

Reviewer: ```suggestion
Several different packages in this repository are orchestrated as yarn workspaces. Important packages include:
```

>  - `core`, which is the React/Redux/Typescript frontend for the simulation engine. 

Reviewer: ```suggestion
 - `core`, which is the React/Redux/TypeScript frontend of hCore
```

>  - `engine`, which is the rust implementation of the simulation engine

Reviewer: ```suggestion
 - `engine` contains the hCore simulation engine, written in Rust. This is a legacy engine that is less powerful than [HASH Engine].
```

>  - `engine-web`, which bundles the rust-based `engine` into a webassembly-backed javascript interface using `wasm-bindgen`.

Reviewer: ```suggestion
 - `engine-web` bundles the `engine` package into a WebAssembly-backed JavaScript interface using `wasm-bindgen`.
```

 Additional utility packages also exist to facilitate minor conveniences.

>  While each package can be built and run separately using the `yarn` commands within its package (see the given package's package.json file for  guidance), in general they'll all be built together by running the `yarn` command from the `sim-core` folder.

Reviewer: ```suggestion
 While each package can be built and run separately using the `yarn` commands within its package (see the given package's `package.json` file for  guidance), in general they'll all be built together by running the `yarn` command from the `sim-core` folder.
```

I don't like saying "in general" here. Do we not always do this? Are there exceptions?

Author: I generally ( ;-) ) only ever run the full 'yarn' at installation, and run yarn serve:core as my dominant operation. I think that when developing, running 'yarn' is the exception rather than the common pattern.

I'll wordsmith a bit and push and update
